nexus/get_info_test.go

```diff
@@ -64,3 +65,123 @@ func TestGetInfoHandleFromClientNoHeader(t *testing.T) {
+	time.Sleep(20 * time.Millisecond)
+
+	if ctx.Err() != nil {
+		return nil, HandlerErrorf(HandlerErrorTypeDownstreamTimeout, "handler exceeded request timeout of %s", options.Header.Get(headerRequestTimeout))
+	}
+
+	return &OperationInfo{
+		ID:    operationID,
+		State: OperationStateCanceled,
+	}, nil
+}
+
+func TestGetInfoHandlerTimeout(t *testing.T) {
+	ctx, client, teardown := setup(t, &asyncWithInfoTimeoutHandler{})
+	defer teardown()
+
+	type testcase struct {
+		name         string
+		timeout      time.Duration
+		setOnHeader  bool
+		setOnContext bool
+		validator    func(t *testing.T, handle *OperationHandle[*LazyValue], info *OperationInfo, err error)
+	}
+	cases := []testcase{
+		{
+			name:         "time_out: set on context",
+			timeout:      1 * time.Millisecond,
+			setOnHeader:  false,
+			setOnContext: true,
+			validator: func(t *testing.T, handle *OperationHandle[*LazyValue], info *OperationInfo, err error) {
+				require.ErrorContains(t, err, "context deadline exceeded")
+			},
+		},
+		{
+			name:         "time_out: set on header",
+			timeout:      1 * time.Millisecond,
+			setOnHeader:  true,
+			setOnContext: false,
+			validator: func(t *testing.T, handle *OperationHandle[*LazyValue], info *OperationInfo, err error) {
+				require.ErrorContains(t, err, "handler exceeded request timeout of 1ms")
+			},
+		},
+		{
+			name:         "time_out: set on context and header",
+			timeout:      1 * time.Millisecond,
+			setOnHeader:  true,
+			setOnContext: true,
+			validator: func(t *testing.T, handle *OperationHandle[*LazyValue], info *OperationInfo, err error) {
+				require.ErrorContains(t, err, "context deadline exceeded")
+			},
+		},
+		{
+			name:         "success: set on context",
+			timeout:      5 * time.Second,
+			setOnHeader:  false,
+			setOnContext: true,
+			validator: func(t *testing.T, handle *OperationHandle[*LazyValue], info *OperationInfo, err error) {
+				require.NoError(t, err)
+				require.Equal(t, handle.ID, info.ID)
+				require.Equal(t, OperationStateCanceled, info.State)
+			},
+		},
+		{
+			name:         "success: set on header",
+			timeout:      5 * time.Second,
+			setOnHeader:  true,
+			setOnContext: false,
+			validator: func(t *testing.T, handle *OperationHandle[*LazyValue], info *OperationInfo, err error) {
+				require.NoError(t, err)
+				require.Equal(t, handle.ID, info.ID)
+				require.Equal(t, OperationStateCanceled, info.State)
+			},
+		},
+		{
+			name:         "success: set on context and header",
+			timeout:      5 * time.Second,
+			setOnHeader:  true,
+			setOnContext: false,
+			validator: func(t *testing.T, handle *OperationHandle[*LazyValue], info *OperationInfo, err error) {
+				require.NoError(t, err)
+				require.Equal(t, handle.ID, info.ID)
+				require.Equal(t, OperationStateCanceled, info.State)
+			},
+		},
```

Comment:
You don't need this coverage, check that if you don't set a context deadline from the client, the server doesn't set a context deadline either.

You can achieve this by sending a header to the client to tell it what to expect.